### ISSUE TITLE
Holes in cadence in the GPX file

### ISSUE THREAD

**Issue by xalexalex** [bailuk/AAT#103]
Hello and thanks for the immensely useful app!

I recently acquired a set of speed/cadence/heartrate sensors and AAT was the first app I tried them with.

Cadence and heart rate are correctly displayed in the cockpits, but in the GPX file there are a bunch of `<cadence>0</cadence>` lines (approximately 75% of all the cadence lines are zero) whereas heart rate works perfectly.

At first I thought the issue was with the sensor, but then I tried it with another android app (wahoo) and the resulting GPX file looks fine. Also, while riding, the AAT cockpit never shows zero cadence while pedaling (and it actually seems to refresh about every 1-2 seconds) so I have no idea where these holes in the GPX file come from.

I might upload a piece of the GPX file, but I think these strava graphs of the same track (on different days) drive the point home.
Top: AAT, bottom: wahoo
![a](https://user-images.githubusercontent.com/16612842/92412749-5b9ccd00-f14d-11ea-9923-c7c823750f04.png)

I am available in case I can help by providing more information.
Cheers,
Alessandro

**Comment by xalexalex** [comment]
Probably fixed by d46c2176c852e9263dae6b7e032599961edd5daf . I currently do not have access to my bike, but will report back as soon as possible.

**Comment by MaxKellermann** [comment]
@xalexalex I don't think so; the problem as described here is that there are `<cadence>0</cadence>` entries when the cadence should be non-zero. My change emits (old) cadence values again even when there is no new value; it fixes the problem that there is no `<cadence>` element at all.
